src/Polysemy/Scoped.hs

```diff
@@ -28,21 +28,21 @@ import Polysemy.Internal.Combinators
 import Polysemy.Internal.Scoped
 import Polysemy.Internal.Tactics
 
--- | Interpreter for 'Scoped', taking a @resource@ allocation function and a
--- parameterized handler for the plain @effect@.
+-- | Construct an interpreter for a higher-order effect wrapped in a 'Scoped',
+-- given a resource allocation function and a parameterized handler for the
+-- plain effect.
 --
--- >>> interpretScopedH withResource scopedHandler
---
--- @withResource@ is a callback function, allowing the user to acquire the
--- resource for each program wrapped by 'scoped' using other effects, with an
--- additional argument that contains the call site parameter passed to 'scoped'.
---
--- @scopedHandler@ is a handler like the one expected by 'interpretH' with an
--- additional parameter that contains the @resource@ allocated by
--- @withResource@.
+-- This combinator is analogous to 'interpretH' in that it allows the handler to
+-- use the 'Tactical' environment and transforms the effect into other effects
+-- on the stack.
 interpretScopedH ::
   ∀ param resource effect r .
+  -- | A callback function that allows the user to acquire a resource for each
```

Comment:
Format for arguments is `-- ^` -- unless Haddock has improved since I last knew it.

Reply:
this is valid: https://haskell-haddock.readthedocs.io/en/latest/markup.html#documenting-parts-of-a-declaration

Reply:
my impression is that the point of `-- ^` is to allow the docstring to be on the same line

Reply:
Well alright then!